### ISSUE TITLE
Conditional fields in grid not working

### ISSUE THREAD

**Issue by JoeriE** [statamic/cms#1384]
Hi,

I read that conditional fields in a grid item are possible now, but I don't seem to get it working. This is my field: 
```  button:
    fields:
      -
        handle: button_label
        field:
          html_type: text
          type: text
          localizable: false
          listable: hidden
          display: 'Button label'
      -
        handle: external_link
        field:
          type: toggle
          localizable: false
          listable: hidden
          display: 'External link'
          width: 25
      -
        handle: entry
        field:
          type: entries
          localizable: false
          listable: hidden
          display: Entry
          max_items: 1
          mode: default
          if:
            external_link: 'equals false'
          width: 75
      -
        handle: url
        field:
          html_type: url
          type: text
          localizable: false
          listable: hidden
          display: URL
          if:
            external_link: 'equals true'
          width: 75
    mode: stacked
    max_rows: 1
    add_row: 'Add button'
    reorderable: false
    type: grid
    localizable: false
    listable: hidden
    display: Button
```

I've also tried: button.external_link and default.button.external_link (default is the fieldset)

Thanks
Joeri

**Comment by benfurfie** [comment]
Just ran into the same issue. This is a blocker to us being able to migrate over from V2.

**Comment by aerni** [comment]
Just bumped into this as well. It's a show stopper. 😕

**Comment by stuffedmotion** [comment]
This is stopping us as well

**Comment by robdekort** [comment]
Possibly linked to #2289

**Comment by JoeriE** [comment]
Any updates on this issue?

**Comment by jesseleite** [comment]
This should be fixed in 3.0.23 by #2767.  Let us know if you're still having trouble and we can re-open!

**Comment by JoeriE** [comment]
@jesseleite Works like a charm, thanks!
